.github/workflows/ci.yml

```diff
@@ -0,0 +1,84 @@
+---
+# The present workflow was made based on the following references:
+# - https://github.com/actions/cache/blob/main/examples.md#haskell---cabal
+# - https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md
+# - https://github.com/haskell/time/blob/master/.github/workflows/ci.yml
+# - https://github.com/stackbuilders/stache/blob/master/.github/workflows/ci.yaml
+name: CI
+
+on: push
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        ghc:
+          - "9.0.1"
+          - "8.10.4"
+          - "8.8.4"
+          - "8.6.5"
+        include:
+          - os: macos-latest
+            ghc: "9.0.1"
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: haskell/actions/setup@v1
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          cabal-version: "3.4"
+      - run: |
+          if [ "$RUNNER_OS" == "Linux" ]
+          then
+            sudo apt-get update
+            sudo apt-get install zsh
+          else
+            brew update
+            brew install zsh
+          fi
+      - run: cabal update
+      - run: cabal configure --enable-tests
+      - run: cabal freeze
+      - uses: actions/cache@v3
+        with:
+          key: v1-${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+          restore-keys: v1-${{ runner.os }}-${{ matrix.ghc }}-
+          path: |
+            ~/.cabal/packages
+            ~/.cabal/store
+            dist-newstyle
+      - run: cabal build --only-dependencies
+      - run: cabal build
+      - run: cabal test
+      - run: |
+          # Fixes issue with different haddock coverage with different ghc versions https://github.com/haskell/haddock/issues/123
+          cabal haddock | grep "100%" | wc -l | grep -E "[4-9]|[1-9][0-9]+" ||
+          (echo "Haddock failed with exit code 1. Have you checked that the minimum of 4 modules with 100% documentation is fulfilled?" && false)
+
+  docker:
```

Comment:
@sestrella This looks nice. Where do you specify that this job will be executed only when a release is happening? :thinking:

Reply:
@CristhianMotoche this job mimics the [existing workflow](https://github.com/stackbuilders/hapistrano/blob/master/.github/workflows/build-and-test.yml#L59-L69), in that sense, the current workflow only builds and runs the image, it doesn't interact with Docker Hub therefore, the `docker/login-action` was not required at all